src/plugins/intel_gpu/tests/test_cases/gemm_gpu_test.cpp

```diff
@@ -167,3115 +214,890 @@ TEST(gemm_gpu, basic_bfyx_t2_inplace_crop_with_pad) {
-    }
-}
-
-TEST(gemm_gpu, basic_input3_t1t2) {
-    auto& engine = get_test_engine();
-    auto input = engine.allocate_memory({ data_types::f32, format::bfyx, { 1, 1, 4, 3 } });
-    auto input2 = engine.allocate_memory({ data_types::f32, format::bfyx, { 1, 1, 3, 2 } });
-    auto input3 = engine.allocate_memory({ data_types::f32, format::bfyx, { 1, 1, 2, 4 } });
-    float alpha = 2.f;
-    float beta = 3.f;
-
-    std::vector<float> input_data = {
-        1.0f, 2.0f, 3.0f, 4.0f,
-        1.0f, 0.0f, 1.0f, 0.0f,
-        0.0f, 0.0f, 0.0f, 0.0f
-    };
-
-    std::vector<float> input_data2 = {
-        3.0f, 3.0f, 1.0f,
-        2.0f, 1.0f, 2.0f,
-    };
-
-    std::vector<float> input3_data = {
-        1.0f, 0.0f,
-        1.0f, 0.0f,
-        2.0f, 2.0f,
-        1.0f, 1.0f,
-    };
+INSTANTIATE_TEST_SUITE_P(
+        GemmGPUTest_t1t2,
+        GemmGPUTest,
+        ::testing::Combine(
+            ::testing::Values(std::vector<std::vector<int32_t>>{{2, 1, 3, 4}, {2, 1, 4, 1}}),
+            ::testing::Values(std::vector<std::vector<float>>{
+                                  {
+                                      1.f, -2.f,  3.f,
+                                      -4.f, 5.f,  6.f,
+                                      1.f, 2.f, 3.f,
+                                      3.f, 2.f, -1.f,
+
+                                      1.f, -2.f,  3.f,
+                                      -4.f, 5.f,  6.f,
+                                      1.f, 2.f, 3.f,
+                                      3.f, 2.f, -1.f,
+                                  },
+                                  {
+                                      2.f, 5.f, -4.f, -7.f,
+                                      2.f, 5.f, -4.f, -7.f,
+                                  }
+                              }),
+            ::testing::ValuesIn(formats),
+            ::testing::ValuesIn(float_types),
+            ::testing::Values(std::vector<float>{
+                                  -43.f, -1.f, 31.f,
+                                  -43.f, -1.f, 31.f
+                              }),
+            ::testing::Values(true),
+            ::testing::Values(true),
+            ::testing::Values(1.0f),
+            ::testing::Values(0.0f)
+            )
+        );
+
+INSTANTIATE_TEST_SUITE_P(
+    GemmGPUTest_basic_input3, GemmGPUTest,
+    ::testing::Combine(
+        ::testing::Values(std::vector<std::vector<int32_t>>{
+            {1, 1, 3, 2}, {1, 1, 2, 3}, {1, 1, 2, 2}}),
+        ::testing::Values(std::vector<std::vector<float>>{
+            {1.0f, 2.0f, 3.0f, 1.0f, 0.0f, 1.0f},
+            {
+                3.0f,
+                3.0f,
+                1.0f,
+                2.0f,
+                1.0f,
+                2.0f,
+            },
+            {
+                1.0f,
+                0.0f,
+                2.0f,
+                0.0f,
+            }}),
+        ::testing::ValuesIn(formats), ::testing::ValuesIn(all_types),
```

Comment:
If we combine all possible combination, There will be too many redundant tests generated .. Please select meaningful subsets to reduce the redundant CI time.
Also,  could you generate random inputs and expected outputs with that from ngraph? 
(Recently we are adding TCs in that way) 
Also, to test blocked format, the shapes should be big enough, for example, bsv16 should be tested for bsize > 16 as well as <16 and fsv16 should be tested for f > 16 as well as < 16